fixes for saddling and pins

Kherae opened the pull request:
fixed debugger to use correct naming
filtered pin removal so it doesn't remove pins it shouldn't. no idea why but it was removing pretty much every pin on the map that wasnt the lox, before this. this also resulted in a severe logspam when used alongside iloxyou, as they were trying to remove each others' pins. it's now filtered to check for the lox trophy icon (which this mod uses for its pin) being the pin icon.
fixed unremovable saddle.
Note: if left idle for an indeterminate amount of time, under certain conditions (hunger?), asksvin will require a remount to regain control of. unsure if vanilla behavior.

AzumattDev reviewed (changes requested):
Just tested these changes. This actually causes more bugs and makes the mount never think you are mounted. Stays frightened, attacks and runs uncontrollably. I appreciate the pull request but this appears to cause more problems than it was going to solve. The pin filter is something I missed. Thank you for that portion though!

Kherae commented:
reverted my 'fixes' (the use of OnUseStop)
fixed the actual issue, which was RidingAsksvin not being set to false BEFORE StopDoodadControls was fired, causing the prefix to skip the original function. 
mirrored these to ILoxYou.

go figure, I wasn't entirely wrong about it being 'missing'...but my implementation of a 'fix' was definitely not the right one for this case.

Kherae commented:
should note, however, that the mount deciding to run away when frightened (which can happen when hit) is vanilla behavior. confirmed that when testing with just vanilla.

AzumattDev commented:
Just adding some comments. Haven't reviewed fully yeet, but it just seems like you're reverting the changes I made recently, which means the bug of attacking comes back. Though some of your code is a bit different, so it might help. Will review later.

This is the changes I mean:
https://github.com/AzumattDev/AsksvinImproved/commit/5dc96eb1ea0f77df0592a68cd8534a1c9dc0aa77

Kherae commented:
as per the discord bug report thread regarding this issue, yes.

AzumattDev commented:
Sorry, hard to talk in two places.
